scipy/differentiate/tests/test_differentiate.py

```diff
@@ -238,6 +238,28 @@ def df(x, p):
         ref = xp.asarray(ref, dtype=xp.asarray(1.).dtype)
         xp_assert_close(res.df, ref)
 
+    def test_initial_step(self, xp):
+        # Test that `initial_step` works as expected and is vectorized
+        def f(x):
+            return xp.exp(x)
+
+        x = xp.asarray(0., dtype=xp.float64)
+        step_direction = xp.asarray([-1, 0, 1])
```

Comment:
Including `step_direction` here because step size is different depending on step direction. Note that the size is along axis 0 and the direction is along axis 1.